app/src/main/java/unipd/se18/ocrcamera/PreProcessing.java

```diff
@@ -0,0 +1,369 @@
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case 2: //If the image is too dark
+                Log.d(TAG, "Case==2 ==> Too dark");
+                //Lightens the colour's brightness until it's in an optimal value
+                for(double changeBrightness=0; changeBrightness!=240; changeBrightness+=15){
+                    //Converts an array to another data type with optional scaling.
+                    bright.convertTo(modifiedMat, -1, 1, changeBrightness);
+                    if(isBright(modifiedMat)==0){
+                        //If the conversion failed it returns the original image
+                        try{
+                            return IPUtils.conversionMatToBitmap(modifiedMat);
+                        } catch (ConversionFailedException error){
+                            Log.e(TAG, error.getErrorMessage());
+                            return image;
+                        }
+                    }
+                }
+                break;
+
+            case 0: //Image is neither too bright nor too dark
+                Log.d(TAG, "Case==0 ==> Perfect image");
+                return image;
+        }
+        return image;
+    }
+
+
+    /**
+     * @author Thomas Porro(g1), Oscar Garrido (g1), Giovanni Fasan(g1).
+     * See PreProcessingMethods.java
+     */
+    @Override
+    public boolean isBlurred(Bitmap image) {
+
+        //Total number of color
+        int maxLap = -16777216;
+
+        //Threshold above which the color is out of focus
+        final int threshold = -6118750;
+
+        //Converts the image into a matrix
+        Mat imageMat;
+        try {
+            imageMat = IPUtils.conversionBitmapToMat(image);
+        } catch (ConversionFailedException error){
+            Log.e(TAG, error.getErrorMessage());
+            return false;
+        }
+
+        //Turn the colored matrix into a grayscale matrix
+        Mat grayImageMat = new Mat();
+        Imgproc.cvtColor(imageMat, grayImageMat, Imgproc.COLOR_BGR2GRAY);
+
+        /*Use the openCV's Laplacian methods to apply a transformation that allow us to detect
+          the image blurriness*/
+        Mat laplacianMat = new Mat();
+        Imgproc.Laplacian(grayImageMat, laplacianMat, CV_8U);
+        Mat laplacianMat8Bit = new Mat();
+        laplacianMat.convertTo(laplacianMat8Bit, CV_8UC1);
+
+        //Create a Bitmap with the given matrix, and obtain all the pixels from it
+        Bitmap laplacianImage;
+        try{
+            laplacianImage = IPUtils.conversionMatToBitmap(laplacianMat8Bit);
+        } catch (ConversionFailedException error){
+            Log.e(TAG, error.getErrorMessage());
+            return false;
+        }
+
+        //TODO we have to comment this lines and add getPixels to the builder
```

Comment:
Is this code for testing? Should I review this? Why are there some "todos" that seem important not done something?